framework/src/timeintegrators/BDF2.C

```diff
@@ -64,6 +65,28 @@ BDF2::computeTimeDerivatives()
   }
 }
 
+void
+BDF2::computeDotProperties(Moose::MaterialDataType type, THREAD_ID tid) const
+{
+  auto material_data = _fe_problem.getMaterialData(type, tid);
+
+  auto props = material_data->getPropertiesForDot();
+  auto old_props = material_data->getOldPropertiesForDot();
+  auto older_props = material_data->getOlderPropertiesForDot();
+  auto dot_props = material_data->getDotProperties();
```

Comment:
All of these should be `auto &`... you are making vector copies!

Reply:
My misunerstanding, thought `auto` will respect `&`.